components/image/slider/src/index.js

```diff
@@ -1,45 +1,73 @@
+import React, {Component} from 'react'
 import PropTypes from 'prop-types'
-import React from 'react'
 import ReactSlidy from 'react-slidy'
+import cloneDeep from 'lodash.clonedeep'
+import cx from 'classnames'
+// TODO: To be used when we can publish packages again to the npm registry.
+// import IconCamera from '@schibstedspain/sui-svgiconset/lib/Camera'
 
-const ImageSlider = (props) => {
-  const slides = getSlides(props.images, props.linkFactory)
+const NO_OP = () => {}
 
-  return (
-    <div onClick={props.handleClick} className='sui-ImageSlider'>
-      { hasMoreThanOneImage(props.images)
-        ? (<ReactSlidy {...props.sliderOptions} dynamicContent={props.dynamicContent}>{ slides }</ReactSlidy>)
-        : slides
-      }
-    </div>
-  )
-}
+class ImageSlider extends Component {
+  constructor (props) {
+    super(props)
+    const {sliderOptions} = props
+    // In order to accept a custom doAfterSlide callback, and to avoid altering props.sliderOptions
+    // sliderOptions is deeply cloned.
+    this._sliderOptions = cloneDeep(sliderOptions)
+    this._sliderOptions.doAfterSlide = (currentSlide) => {
+      this.setState(currentSlide)
+      sliderOptions.doAfterSlide && sliderOptions.doAfterSlide(currentSlide)
+    }
+  }
+
+  state = {currentSlide: 0}
+
+  render () {
+    const {images, linkFactory, handleClick, dynamicContent, enableCounter} = this.props
+    const slides = this._getSlides(images, linkFactory)
+    return (
+      (slides.length > 0) &&
+      <div onClick={handleClick} className='sui-ImageSlider'>
+        { (slides.length > 1)
+          ? (<ReactSlidy {...this._sliderOptions} dynamicContent={dynamicContent}>{ slides }</ReactSlidy>)
+          : slides
+        }
+        {enableCounter && this._buildCounter(slides.length)}
+      </div>
+    )
+  }
 
-/**
- * Says if there are more than one image in the given list.
- * Implemented for semantic purposes.
- * @param {Array} images List given by props.images.
- * @return {boolean}
- */
-const hasMoreThanOneImage = (images) => (images && images.length > 1)
+  _buildCounter (totalImages) {
+    const classNames = cx('sui-ImageSlider-counter', `sui-ImageSlider-counter--${this.props.counterPosition}`)
+    const Icon = this.props.counterIcon
+    return <div className={classNames}>
+      <Icon svgClass='sui-ImageSlider-counterIcon' />
+      <span className='sui-ImageSlider-counterText'>{this.state.currentSlide + 1}/{totalImages}</span>
```

Comment:
Just one suggestion, but it could be so cool to have here some kind of customizable pattern for this string, for example, having a prop for the counter data named `counterData`:
```javascript
<ImageSlider counterData={{
  patternFactory: ({ current, separator = '/', total }) => `${current}${separator}${total}`,
  separatorValue: '/'
}} />
```
And then, inside the component:
```javascript
 <span className='sui-ImageSlider-counterText'>
  {counterData.patternFactory({
     current: this.state.currentSlide + 1,
     total: totalImages
  })}
</span>
```

Reply:
I love it but should I? I mean, this component is supposed to be sui compliant (definition on WIP) (I could use a troll/lol face here) Should I open that door?

Reply:
Aaaaand I did it anyway 👍